rasa/shared/utils/domain_loading.py

```diff
@@ -0,0 +1,198 @@
+from typing import Dict, Text, List, Any, NamedTuple, Tuple
+
+from rasa.shared.constants import (
+    SESSION_CONFIG_KEY,
+    KEY_INTENTS,
+    KEY_ENTITIES,
+    KEY_FORMS,
+    KEY_ACTIONS,
+    KEY_E2E_ACTIONS,
+    KEY_RESPONSES,
+    KEY_SLOTS,
+    DEFAULT_SESSION_EXPIRATION_TIME_IN_MINUTES,
+    DEFAULT_CARRY_OVER_SLOTS_TO_NEW_SESSION,
+    LATEST_TRAINING_DATA_FORMAT_VERSION,
+)
+from rasa.shared.utils.validation import KEY_TRAINING_DATA_FORMAT_VERSION
+
+
+class SessionConfig(NamedTuple):
+    """The Session Configuration."""
+
+    session_expiration_time: float  # in minutes
+    carry_over_slots: bool
+
+    @staticmethod
+    def default() -> "SessionConfig":
+        """Returns the SessionConfig with the default values."""
+        return SessionConfig(
+            DEFAULT_SESSION_EXPIRATION_TIME_IN_MINUTES,
+            DEFAULT_CARRY_OVER_SLOTS_TO_NEW_SESSION,
+        )
+
+    def are_sessions_enabled(self) -> bool:
+        """Returns a boolean value depending on the value of session_expiration_time."""
+        return self.session_expiration_time > 0
+
+    def as_dict(self) -> Dict:
+        """Return serialized `SessionConfig`."""
+        return {
+            "session_expiration_time": self.session_expiration_time,
+            "carry_over_slots_to_new_session": self.carry_over_slots,
+        }
+
+
+def extract_duplicates(list1: List[Any], list2: List[Any]) -> List[Any]:
```

Comment:
I would either keep this stuff in the `Domain` class (as private methods) if it's specific to domains or put it in a general utility file if it can/will be used elsewhere (unit-tested independently). I don't think we need a `domain_loading` utility module.

Reply:
Done 👍🏼 Seems the functions I moved into `common.py` have 100% test coverage, do you think I should still include some unit tests in `test_common.py`?

Reply:
Yeah I still think it's worth unit testing them (at least a bit ) as they live alone now.